pysparkling/fileio/fs/local.py

```diff
@@ -13,65 +13,54 @@
 
 class Local(FileSystem):
     def __init__(self, file_name):
-        FileSystem.__init__(self, file_name)
+        super(Local, self).__init__(file_name)
 
     @staticmethod
     def resolve_filenames(expr):
         if expr.startswith('file://'):
             expr = expr[7:]
-
         if os.path.isfile(expr):
             return [expr]
 
-        files = []
         t = Tokenizer(expr)
         prefix = t.next(['*', '?'])
-        for root, dirnames, filenames in os.walk(prefix):
-            root_wo_slash = root[:-1] if root.endswith('/') else root
+        files = []
+        for root, _, filenames in os.walk(prefix):
             for filename in filenames:
-                if fnmatch.fnmatch(root_wo_slash+'/'+filename,
-                                   expr) or \
-                   fnmatch.fnmatch(root_wo_slash+'/'+filename,
-                                   expr+'/part*'):
-                    files.append(root_wo_slash+'/'+filename)
-        # files += glob.glob(expr)+glob.glob(expr+'/part*')
-
+                path = os.path.join(root, filename)
+                if fnmatch(path, expr) or fnmatch(path, expr + '/part*'):
+                    files.append(path)
         return files
 
+    @property
+    def file_path(self):
+        if self.file_name.startswith('file://'):
+            return self.file_name[7:]
+        return self.file_name
+
     def exists(self):
-        path_local = self.file_name
-        if path_local.startswith('file://'):
-            path_local = path_local[7:]
-        return os.path.exists(path_local) or os.path.exists(path_local+'/')
+        return os.path.exists(self.file_path)
```

Comment:
`self.file_path` is a property here, to avoid repeating the code that strips the `file://`.

The only test that I actually removed is when you add a `/` to the path, because `os.path` handles that already.